### PR TITLE
tests: add "core" suite for UC specific tests

### PR DESCRIPTION
**Pull request by anonymouse64**
This will make it easier to ensure we have test coverage of all the Ubuntu Core variants going forward.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -663,20 +663,9 @@ suites:
...
-        prepare-each: |
-            "$TESTSLIB"/prepare-restore.sh --prepare-suite-each
-        restore-each: |
-            "$TESTSLIB"/prepare-restore.sh --restore-suite-each
-        restore: |
-            "$TESTSLIB"/prepare-restore.sh --restore-suite
-    tests/core20/:
-        summary: Subset of core20 specific tests
-        systems: [ubuntu-core-20-*]
+    tests/core/:
```
Comment (zyga) [inline spread.yaml]: Unless I'm missing something the suite directory is not renamed. This should probably be included in the patch.

Comment (zyga) [reply spread.yaml]: Eh, I'm blind :D Sorry, github UI for renames is subtle.

**Comment by zyga** [comment]
The failures look relevant:
```
2020-02-07 02:02:14 Error preparing google:ubuntu-core-16-64:tests/core/ : /bin/bash: line 2: cd: /home/gopath/src/github.com/snapcore/snapd/tests/core/snapd-failover: No such file or directory
```

**Comment by anonymouse64** [comment]
Alright fixed this, it was a silly auto exclusion that rsync does with the `-C` option, it always ignores "core" files/dirs because CVS ignores those.

File: tests/lib/prepare.sh [hunk 2]
```diff
@@ -800,12 +800,18 @@ EOF
...
+        # are core dumps, but we have a test suite named "core", so including 
+        # this here will ensure that portion of the git tree is included in the
+        # image
         rsync -a -C \
           --exclude '*.a' \
           --exclude '*.deb' \
           --exclude /gopath/.cache/ \
           --exclude /gopath/bin/govendor \
           --exclude /gopath/pkg/ \
+          --include core \
```
Comment (sergiocazzolato) [inline tests/lib/prepare.sh]: perhaps --include core/ works better to make sure it is matching a dir called core, rigth?

Comment (anonymouse64) [reply tests/lib/prepare.sh]: Done

File: tests/lib/prepare.sh [hunk 3]
```diff
@@ -800,12 +800,18 @@ EOF
     # - golang archive files and built packages dir
     # - govendor .cache directory and the binary,
     if is_core16_system || is_core18_system; then
+        # we need to include "core" here because -C option says to ignore 
+        # files the way CVS(?!) does, so it ignores files named "core" which
+        # are core dumps, but we have a test suite named "core", so including 
+        # this here will ensure that portion of the git tree is included in the
+        # image
```
Comment (mvo5) [inline tests/lib/prepare.sh]: Nice comment, I bet this took a little bit to find out :)

**Comment by anonymouse64** [comment]
The spread failures here were from running snapd-snap specific tests on UC16, which doesn't work.

**Pull request merged**